File: server/src/tripscribe.Dal/Models/Trip.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Npgsql.Replication.PgOutput.Messages;

namespace tripscribe.Dal.Models;

[Table("trips")]
public class Trip
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("title")] public string Title { get; set; }
    
    [Column("description")] public string Description { get; set; }

    [Column("timestamp")] public DateTime Timestamp { get; set; }
    
    [Column("public_view")] public Boolean PublicView { get; set; }
    
    public ICollection<Stop> TripStops { get; set; }

    public ICollection<AccountTrip> AccountTrips { get; set; }
    
    public ICollection<TripReview> TripReviews { get; set; }
}

File: ui/src/types/trip.ts
import { Stop } from "./stop";
import { Account } from "./account";

export type Trip = {
  id?: string;
  title: string;
  description: string;
  timestamp?: Date;
  publicView: boolean;
  accounts?: Account[];
  tripStartDate?: Date;
  tripEndDate?: Date;
  stops?: Stop[];
};
